### ISSUE TITLE
Training loss of rpn doesn't converge in stage1

### ISSUE THREAD

**Issue by imbinwang** [rbgirshick/py-faster-rcnn#84]
I have trained faster-rcnn on my own dataset by faster_rcnn_alt_opt, while the training loss of rpn doesn't converge in stage1. The loss curve shows as following.

![loss_curve_alt_opt](https://cloud.githubusercontent.com/assets/8622373/13134688/3c666d7a-d644-11e5-9897-7ecfbd8fe82c.png)

My dataset consists of 2 classes(including background) and the image size is 640x480. All training configuration parameters are default (except cfg.TRAIN.SCALES and cfg.TEST.SCALES, the scales are set as [480]). 

I wonder what the loss curve looks like when training on PASCAL VOC dataset. Did anyone get the same problem when training rpn on new dataset?

Thanks.

**Comment by imbinwang** [comment]
@rbgirshick Could you show me  the loss curves when you trained the rpn on PASCAL VOC dataset in stage1? I got invalid loss(log loss) curves when training on my own dataset (as the figure shows).

Thank you.

**Comment by eternautaCAT** [comment]
I'm training with 2 classes two and i get weird loss curves too:

![figure_1](https://cloud.githubusercontent.com/assets/6147283/13826768/e9bf19c8-eb97-11e5-9060-9fb9e1caa24a.png)

This is the loss curve at 12000 iterations (the 600 is because i print the loss every 20 iterations).

Any idea about how could be wrong?

**Comment by Austriker** [comment]
@imbinwang @ericromanenghi have you solved your issue ? I am also trying to train on a one class dataset (+background) but the rpn_bbox_loss is really unstable (between 12 and 0.03)

**Comment by mlliu** [comment]
@imbinwang @ericromanenghi same error too, so sad .....
![image](https://cloud.githubusercontent.com/assets/16427341/19725776/2118cec8-9bba-11e6-9ded-02b21754e3e1.png)

**Comment by Austriker** [comment]
@mlliu In the end i think the answer is that it's normal. The loss spikes when the net trains on a really new  image. In the end even with this loss curve I have really good results. Check the test results.

**Comment by public1024** [comment]
@mlliu hi bro, sorry to bother you, but my loss of brpn_loss_bbox become nan very quickly, do you have any idea about that?

**Comment by public1024** [comment]
@mlliu hi bro, sorry to bother you, but my loss of brpn_loss_bbox become nan very quickly, do you have any idea about that?
